### ISSUE TITLE
Improve 'ALL' search by auto excluding value types

### ISSUE THREAD

**Issue by Symbai** [cheat-engine/cheat-engine#2918]
If I search for any value higher than 255 I still get 'byte' results. Same for other value types. I suggest to automatically exclude those types where the search value exceeds the min/max value. This would not only speed up the (re)scan a LOT but also reduces the noise.

The reason is quite simple. Most of the time when we search for 'ALL' its because we dont know if the value is stored as 4 bytes / float or double. But we still get plenty of byte and 2 bytes results which makes no sense.

**Comment by Banaanae** [comment]
Something like multisearch, (either aswell or instead of) where we can choose multiple data types

**Comment by Symbai** [comment]
> where we can choose multiple data types

That wouldn't be "auto excluding" like I suggested. And that would be a NEW search type while I suggest to modify an existing ("all") search type.

**Comment by Banaanae** [comment]
I know, just throwing around similar ideas

**Comment by cheat-engine** [comment]
> Something like multisearch, (either aswell or instead of) where we can choose multiple data types

Still the all type.  Go to settings->scan settings and tick the fields you like for the all type

**Comment by Banaanae** [comment]
Very useful :)

Also helps OP with 
> Most of the time when we search for 'ALL' its because we dont know if the value is stored as 4 bytes / float or double.

**Comment by Symbai** [comment]
> Also helps OP with

Not really. Having to change settings for every value I search is dump. It's not high science to know which value fits into which value type. Cheat Engine could and should do this by itself. It feels like a bug to show me byte values as a result for a search value like 12345.

**Comment by cheat-engine** [comment]
But what about people that don't understand big endian, and just scan for a value like 500 ?  The all type with byte checked, would find the correct address, though show it differently

**Comment by Symbai** [comment]
How many times do you think this would be the case versus not the case?
